spring-cloud-gcp-data-firestore/src/main/java/org/springframework/cloud/gcp/data/firestore/repository/query/PartTreeFirestoreQuery.java

```diff
@@ -86,14 +90,26 @@ public Object execute(Object[] parameters) {
 		}
 
 		StructuredQuery.Builder builder = createBuilderWithFilter(parameters);
-		Flux<?> result = this.reactiveOperations.execute(builder, this.persistentEntity.getType());
 
+		// Handle Pageable parameters.
+		if (!getQueryMethod().getParameters().isEmpty()) {
+			ParameterAccessor paramAccessor = new ParametersParameterAccessor(getQueryMethod().getParameters(),
+					parameters);
+			Pageable pageable = paramAccessor.getPageable();
+			if (pageable != null && pageable.isPaged()) {
```

Comment:
What about Slice query support, like we have in Datastore? Does it make sense for Firestore too?

Reply:
See above comments, I think slice/page support may not make sense for a reactive repository.